### ISSUE TITLE
Catch-all events

### ISSUE THREAD

**Issue by oswdr** [mateodelnorte/sourced#19]
Hi Matt,

I was trying out your library, works great. I like the simplicity! 

However I wan't to catch all events using one single event handler, but it seems it currently only supports catching events on instances of Entities. Is this by design?

Many thanks,

Joey

**Comment by eddieajau** [comment]
Do you mean to, for example, be able to log all events?

**Comment by mateodelnorte** [comment]
It's by design in that Entities publish events by inheriting the behavior of Node's standard EventEmitter class. 

If I were to do this, I would just `.enqueue('eventName', this.toJson());` both an event which describes exactly what action was just performed on an entity, as well as a more generic `.enqueue('updated', this.toJson());` event. Then, you can listen for both the specific event or all. It would look something like: 
```
Invoice.prototype.approved = function (param) {
  this.approved = true;
  this.approvedAt = param.approvedAt;
  this.approvedBy = param.approvedBy;
  this.enqueue('approved', this.toJson());
  this.enqueue('updated', this.toJson());
};
```

If that's not fancy enough for you (I tend to prefer code that is less fancy, and more obvious) you could combine the top two answers from [here](http://stackoverflow.com/questions/5178869/listen-to-all-emitted-events-in-node-js) and monkey patch your entities to use EventEmitter2, which would allow you to use wildcards to determine which events fire your listeners. You could create a little factory function which instantiates your entity, monkey patches it, and gives it back to you - ready to roll.

**Comment by oswdr** [comment]
Thanks for you reply.

What I meant is if it is possible to catch all events on for example on the repository. This makes it possible to have my read services listen to all events from all entities. Please, if this is bad design - let me know.

More specifically, current situation:
`
var repository = ...
var card = new Card();
card.initialize(cmd);
card.on('initialize', () => ...);
repository.commit(card, () => res.send());
`

Preferred situation:
`
var repository = ...
var card = new Card();
card.initialize(cmd);
repository.commit(card, () => res.send());
repository.on('initialize', () => ...);
`

**Comment by mateodelnorte** [comment]
The top scenario you have is preferred, as it's the actual entity being
initialized, not the repo. Note the difference between an Entity's `.emit`
and `.enqueue`. enqueue will do exactly as emit, but only after the entity
has been committed to the repo. That may be the missing part you're looking
for.

**Comment by oswdr** [comment]
Thanks for the quick reply, one last question :-). How would my read side respond on these events? In other words, how would they know an event was thrown?

**Comment by mateodelnorte** [comment]
You'll publish a message to a read-side process, which will instruct it to
update your read data store. Look at servicebus for that.

**Comment by oswdr** [comment]
I will look into it, thanks!
